lib/engines/content_object_store/app/lib/content_object_store/publishable.rb

```diff
@@ -30,6 +30,17 @@ def publish_with_rollback(schema:, title:, details:)
       end
     end
 
+    def schedule_with_rollback
+      raise ArgumentError, "Local database changes not given" unless block_given?
+
+      ActiveRecord::Base.transaction do
+        content_block_edition = yield
+
+        content_block_edition.public_send(:schedule!)
+        ContentObjectStore::SchedulePublishingWorker.queue(content_block_edition)
```

Comment:
ah this was something I wanted to bring up - I'm not sure what we would do if the job fails, and I'm not sure what Whitehall do beyond looking at the Sidekiq dashboard manually. Would this be a subsequent ticket to investigate?

Reply:
we chatted about this and TLDR:
* if the function to publish a scheduled Edition fails, the sidekiq admin dashboard and Grafana will catch that and retry the job, will be shown in  https://docs.publishing.service.gov.uk/manual/sidekiq.html#sidekiq-on-gov-uk 
* I don't know how we are alerted to errors on Whitehall. We think Sentry is used, but not sure where alerts go, and it would be good to have Object Store specific ones in our dev channel if possible. Created a ticket here https://trello.com/c/K19zXzAU/469-tech-task-sentry-for-the-object-store